OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -685,15 +694,36 @@ protected virtual void OnBecomingIdle(Actor self)
 				}
 			}
 
-			var isCircler = !Info.FlightDynamics.HasFlag(FlightDynamic.Hover);
-			if (!atLandAltitude && Info.LandWhenIdle && Info.LandableTerrainTypes.Count > 0)
+			if ((atLandAltitude && !CanLand(self.Location) && ReservedActor == null)
```

Comment:
This check is a monster. Would it simplify things at all if this was moved to the bottom as an `else` (or at worst a simpler `else if`)?

Reply:
Simplified the entire block as much as I could.